autofill-observer-demo/script.js

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ----------------------------- AUTOFILL LOGIC ----------------------------- //
+
+// Possible values for autofill statuses
+const AUTOFILLED = 'autofilled';
+const NEVER_AUTOFILLED = 'never-autofilled';
+const AUTOFILLED_THEN_FIXED = 'autofilled-then-fixed';
+
+// Global variable storing autofill statuses for each field
+const autofillStatuses = {};
+// Example: {
+//     "name": "autofilled",
+//     "street-address": "never-autofilled",
+//     "country": "autofilled-then-fixed"
+// }
+
+function initializeAutofillObserver(formEl) {
```

Comment:
Should this be initializeChangeObservers?

Reply:
Sounds good, renamed.